flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/AbstractKeyedCEPPatternOperator.java

```diff
@@ -126,6 +129,7 @@ public AbstractKeyedCEPPatternOperator(
 		this.isProcessingTime = Preconditions.checkNotNull(isProcessingTime);
 		this.keySerializer = Preconditions.checkNotNull(keySerializer);
 		this.nfaFactory = Preconditions.checkNotNull(nfaFactory);
+		this.conditionRegistry = nfaFactory.createNFA().getConditionRegistry();
```

Comment:
I am ok that `NFA` right now has pointer to `conditionRegistry`. 

The thing I meant is just in this line, as we call `createNFA()` and ignore the created `NFA`. The `nfaFactory` also has a pointer to the same `nfaRegistry` so instead of `nfaFactory.createNFA().getConditionRegistry()` we could call `nfaFactory.getConditionRegistry()`. What do you think?